### PR TITLE
Update how_to_test_systems.rs

### PR DESCRIPTION
**Pull request by Thewsthews**
Objective

- The PR introduces small adjustments and substitutes.

Solution

- A cleaner loop method is introduced, and one of the functions is optimised accordingly. The explicit for loop is replaced with the iter_mut() method which makes it more efficient.

### CODE REVIEW

File: tests/how_to_test_systems.rs [hunk 1]
```diff
@@ -32,14 +32,12 @@ fn despawn_dead_enemies(
...
-    for mut enemy in &mut enemies {
-        enemy.hit_points -= 1;
-    }
+fn hurt_enemies(mut enemies: Query<&mut Enemy>){
+    enemies.iter_mut().for_each(|mut enemy| enemy.hit_points -= 1);
 }
 
-fn spawn_enemy(mut commands: Commands, keyboard_input: Res<ButtonInput<KeyCode>>) {
-    if keyboard_input.just_pressed(KeyCode::Space) {
+fn spawn_enemy(mut commands: Commands, keyboard_input: Res<Input<KeyCode>>) {
```
Comment (alice-i-cecile) [inline tests/how_to_test_systems.rs:39]: I don't think this change compiles.

File: tests/how_to_test_systems.rs [hunk 2]
```diff
@@ -32,14 +32,12 @@ fn despawn_dead_enemies(
     }
 }
 
-fn hurt_enemies(mut enemies: Query<&mut Enemy>) {
-    for mut enemy in &mut enemies {
-        enemy.hit_points -= 1;
-    }
+fn hurt_enemies(mut enemies: Query<&mut Enemy>){
+    enemies.iter_mut().for_each(|mut enemy| enemy.hit_points -= 1);
```
Comment (alice-i-cecile) [inline tests/how_to_test_systems.rs:36]: This is fine, but I don't think it's a clear improvement when the point of the example is to teach something else.

File: tests/how_to_test_systems.rs (hunk 1 above)
Comment (Thewsthews) [reply tests/how_to_test_systems.rs:39]: I was of the idea that the tweak being to the generic type parameter would not cause any compile errors

Comment (alice-i-cecile) [reply tests/how_to_test_systems.rs:39]: Yeah, this was renamed from `Input` to `ButtonInput` in Bevy 0.13 or so :)

**Comment by alice-i-cecile** [comment]
Closing this out, but do say hi on Discord if you'd like help finding issues to work on :) The [Contributing Guide](https://bevyengine.org/learn/contribute/introduction/) is also quite useful!

**Pull request closed without merging**